docs/customizations/header.md

```diff
@@ -0,0 +1,301 @@
+# Header
+
+## Table of contents
+
+- [Interfaces](#interfaces)
+  - [IHeaderProps](#iheaderprops)
+  - [IHeaderComponentOverrides](#iheadercomponentoverrides)
+  - [IHeaderControlProps](#iheadercontrolprops)
+  - [IHeaderStyleProps](#iheaderstyleprops)
+- [Sample Scenarios](#sample-scenarios)
+  - [Changing header title and icon](#changing-header-title-and-icon)
+  - [Changing button icons](#changing-button-icons)
+  - [Hiding sub-components](#hiding-sub-components)
+  - [Adding a custom button](#adding-a-custom-button)
+  - [Adding a custom image](#adding-a-custom-image)
+  - [Changing element styles](#changing-element-styles)
+
+## Interfaces
+
+### [IHeaderProps](https://github.com/microsoft/omnichannel-chat-widget/blob/main/chat-components/src/components/header/interfaces/IHeaderProps.ts)
+
+The top-level interface for customizing `Header`.
+
+| Attribute | Type | Required | Description | Default |
+| - | - | - | - | - |
+| componentOverrides     | [IHeaderComponentOverrides](#iheadercomponentoverrides)     | No | Used for overriding default `Header` components, e.g., icon, title, minimize button, and close button | -
+controlProps | [IHeaderControlProps](#iheadercontrolprops) | No | Properties that control the element behariors | -
+styleProps | [IHeaderStyleProps](iheaderstyleprops) | No | Properties that control the element styles | -
+
+### [IHeaderComponentOverrides](https://github.com/microsoft/omnichannel-chat-widget/blob/main/chat-components/src/components/header/interfaces/IHeaderComponentOverrides.ts)
+
+Custom React components can be passed as input to override the default sub-components. Alternatively, you can stringify the React component before passing it in. The `chat-components` library provides one util function that can be used: [encodeComponentString](https://github.com/microsoft/omnichannel-chat-widget/blob/main/chat-components/src/common/encodeComponentString.ts).
```

Comment:
missing default for headerIcon and headerTitle

https://github.com/microsoft/omnichannel-chat-widget/blob/main/chat-components/src/components/header/common/defaultProps/defaultHeaderComponentOverrides.tsx

Reply:
They are for storybooks only. In real code it's not used